openhub/data.py

```diff
@@ -51,5 +51,7 @@ def get_data(for_what):
             except Exception as ex:
                 logger.error(ex)
                 break
+        if type(data) is not list:
+            data = [data]
```

Comment:
Hmm, I'm not sure, but IMO it's not the best way to fix this problem. Under which circumstance is `data` a dict? `data = get_affiliated_committers_data(json_object)`, `data = get_outside_committers_data(json_object)`, `data = get_outside_projects_data(json_object)`, or `data = get_portfolio_projects_data(json_object)`? Maybe fix the problem in the corresponding try catch section. It could be better to fix the problem in function implementation and add docstring to indicate that return type should be a list.

Reply:
It is happening in case of `data = get_affiliated_committers_data(json_object)` and `data = get_outside_projects_data(json_object)` for now, but both were returning list before.

I noticed that in both of the cases only one `affiliated_committer` or `outside_project` is being left on the last page and that data is being returned as `dict`.

So, it may happen with other cases too that a new `outside_committer` or `portfolio_project` gets added and being fetched from the last page then the data would be dict. That's why I did it for all the case to avoid future errors.

> It could be better to fix the problem in function implementation and add docstring to indicate that return type should be a list.

Sounds good, so, I should implement it in all the functions, right?

Reply:
> Sounds good, so, I should implement it in all the functions, right?

Sounds good to me 😛

Reply:
done

Reply:
Firstly, please revise the code so that it checks that this is *only* happening on the last item in the list.

We do not want it happening randomly throughout the data.

Prove it is the last one only.

And once it is proved, we should file an issue upstream to get it fixed, or get a better explanation about why it is happening.

Then add a comment in the source code with a link to the upstream bug.
Then and only then can we merge something like this.

Reply:
No, it is not happening on the last item in the list. 
It is happening when we have only one item on a page.
E.g.: On [page-1](https://www.openhub.net/orgs/coala/affiliated_committers.xml?&page=1) we have lots of items so, our returned [data](https://github.com/coala/community/blob/master/openhub/affiliated_committers.py#L8) is:
```
[{'level': '7', 'kudos': '9', 'name': 'John Vandenberg', 'most_recent_commit': {'project': 'coala-mobans', 'date': 'Jul 2018'}, 'most_commits': {'project': 'Pywikibot', 'commits': '835'}}, {'level': '3', 'kudos': '8', 'name': 'li-boxuan', 'most_recent_commit': {'project': 'gh-board', 'date': 'Jul 2018'}, 'most_commits': {'project': 'tutoria', 'commits': '122'}}, {'level': '4', 'kudos': '8', 'name': 'bekicot', 'most_recent_commit': {'project': 'git-task-list', 'date': 'Jun 2018'}, 'most_commits': {'project': 'git-task-list', 'commits': '40'}}, {'level': '2', 'kudos': '8', 'name': 'nkprince007', 'most_recent_commit': {'project': 'gitmate-2', 'date': 'Jul 2018'}, 'most_commits': {'project': 'gitmate-2', 'commits': '306'}}, {'level': '5', 'kudos': '8', 'name': 'Muhammad Kaisar Arkhan', 'most_recent_commit': {'project': 'gitmate-2', 'date': 'Jun 2018'}, 'most_commits': {'project': 'loklak_server', 'commits': '100'}}, {'level': '2', 'kudos': '8', 'name': 'Sangam Kumar', 'most_recent_commit': {'project': 'cEPs', 'date': 'Jun 2018'}, 'most_commits': {'project': 'coala-bears', 'commits': '6'}}, {'level': '3', 'kudos': '7', 'name': 'Viresh Gupta', 'most_recent_commit': {'project': 'coala-quickstart', 'date': 'May 2018'}, 'most_commits': {'project': 'coala-bears', 'commits': '8'}}, {'level': '2', 'kudos': '7', 'name': 'RaiVaibhav', 'most_recent_commit': {'project': 'coala-mobans', 'date': 'Jun 2018'}, 'most_commits': {'project': 'OpenGenus Cosmos', 'commits': '4'}}, {'level': '3', 'kudos': '7', 'name': 'Nitanshu', 'most_recent_commit': {'project': 'err', 'date': 'Jul 2018'}, 'most_commits': {'project': 'rtfd-cli', 'commits': '72'}}, {'level': '3', 'kudos': '7', 'name': 'Rahul Jha', 'most_recent_commit': {'project': 'python-vulture', 'date': 'Jul 2018'}, 'most_commits': {'project': 'python-vulture', 'commits': '45'}}, {'level': '4', 'kudos': '7', 'name': 'Guangyuan Yang', 'most_recent_commit': {'project': 'FreeBSD Documentation', 'date': 'Jul 2018'}, 'most_commits': {'project': 'FreeBSD Documentation', 'commits': '39'}}, {'level': '3', 'kudos': '7', 'name': 'anctartica', 'most_recent_commit': {'project': 'cib', 'date': 'Jul 2018'}, 'most_commits': {'project': 'cib', 'commits': '13'}}, {'level': '2', 'kudos': '7', 'name': 'MacBox7', 'most_recent_commit': {'project': 'coala-bears', 'date': 'Jun 2018'}, 'most_commits': {'project': 'coala-bears', 'commits': '5'}}, {'level': '3', 'kudos': '7', 'name': 'pareksha', 'most_recent_commit': {'project': 'coala-bears', 'date': 'Jul 2018'}, 'most_commits': {'project': 'coala-bears', 'commits': '3'}}, {'level': '4', 'kudos': '7', 'name': 'palash25', 'most_recent_commit': {'project': 'foaas-cli', 'date': 'Jun 2018'}, 'most_commits': {'project': 'foaas-cli', 'commits': '18'}}, {'level': '2', 'kudos': '7', 'name': 'Kriti Rohilla', 'most_recent_commit': {'project': 'coala-bears', 'date': 'Jul 2018'}, 'most_commits': {'project': 'coala-bears', 'commits': '3'}}, {'level': '3', 'kudos': '7', 'name': 'Ishan Srivastava', 'most_recent_commit': {'project': 'coala-quickstart', 'date': 'Jul 2018'}, 'most_commits': {'project': 'coala-quickstart', 'commits': '14'}}, {'level': '1', 'kudos': '7', 'name': 'Monal5031', 'most_recent_commit': {'project': 'dependency_management', 'date': 'Jul 2017'}, 'most_commits': {'project': 'dependency_management', 'commits': '2'}}, {'level': '1', 'kudos': '7', 'name': 'manankalra', 'most_recent_commit': {'project': 'coala-bears', 'date': 'Jan 2018'}, 'most_commits': {'project': 'coala-bears', 'commits': '1'}}, {'level': '1', 'kudos': '1', 'name': 'Shrikrishna Singh', 'most_recent_commit': {'project': 'coala-community', 'date': 'Jan 2018'}, 'most_commits': {'project': 'coala-community', 'commits': '1'}}]
```
Which is a list of all the items on that page.

But when we extract [page-2 data](https://www.openhub.net/orgs/coala/affiliated_committers.xml?&page=2) which has only one item, our [data](https://github.com/coala/community/blob/master/openhub/affiliated_committers.py#L8) is:
```
{'level': '1', 'kudos': '1', 'name': 'Kilari Teja', 'most_recent_commit': {'project': 'coala-projects', 'date': 'May 2018'}, 'most_commits': {'project': 'coala-vs-code', 'commits': '5'}}
```
Which is a dict of that item.

The api on both the pages provides similar format of xml, there is no upstream problem.

I think we can re-vise our code to:
```
If the data contains only one item convert it to list
```
?

Reply:
So, if we get a list, we assert it has more than one member